### DISCUSSION TITLE
NetBox API with Pynetbox, Juptyer and HTTP/HTTPS

### DISCUSSION THREAD

**Discussion by geoff-70** [netbox-community/netbox#9021] [Q&A]
Hello all, trying to go through the Network to code course with Jeremy, I am having trouble with the first "lab" in that with pynetbox within Jupyter Notebooks. I have setup Netbox to use HTTPS and installed certificate etc (from read the docs) The lab calls for API HTTP calls but my system will not allow that over HTTPS and throws an error. I would of thought in a production environment that HTTPS would be wise (I am using a lab but learning this for  prod network) Anyhow, do I turn off HTTPS or is there some other way to connect to Netbox API to create objects? Sample code below. 

import pynetbox

nb = pynetbox.api(
    'http://192.168.189.164/',
    token='13bdb0c38d660fe4bfdca8a6a6e3aa867222ff7f')

chs = nb.dcim.sites.create(
      name='Charleston',
      slug='chs'
)

Here is one of the errors:
The requested url: http://192.168.189.164/api/dcim/sites/ could not be found
So essentially I cannot get to my site via HTTP, how do I fix this or do I want to given my point above about using HTTPS, if so, how can I run the code above over HTTPS as it fails also. Thanks - Geoff

**Comment by geoff-70** [comment]
OK, I have at least found this to workaround the SSL problems for now until I figure out how to use self-signed certs. Use the nb.http_session.verify = False to stop SSL checking when querying netbox. 
nb = pynetbox.api(url='https://192.168.189.164/',token='14cbb61b5cd77b060298c404aca54b34a0f46693')
nb.http_session.verify = False

**Reply by markkuleinio** [reply]
See also https://stackoverflow.com/questions/30405867/how-to-get-python-requests-to-trust-a-self-signed-ssl-certificate

**Reply by geoff-70** [reply]
Hello, thanks for this. I have spent most of the morning trying to fix this issue. I have finally figured out how to copy a .crt file to the 
ca-certificates file via the correct process. Still, I have no cert in my browser. I am not 100% certain now what I am trying to do lol. Is it pynetbox having issues, is it my browser that doesn't trust my local netbox instance? Is it both?

**Reply by geoff-70** [reply]
Hello, update on this issue for those interested! If you follow the guide when installing netbox you can use self-signed certs to obtain a padlock in your browser, albeit with an exception, it works. Next step is pynetbox. When creating your self signed certs you need to add your IP of your netbox server for the Common Name. So mine was 192.168.41.138. You also need to edit /etc/apache2/sites-enabled/netbox.conf and add the same IP to
Change this to your servers name
   ServerName 192.168.41. 138
Then restart Apache, and try your code. You will STILL get an error as follows:
Certificate for 192.168.41.138 has no `subjectAltName`, falling back to check for a `commonName` for now. This feature is being removed by major browsers and deprecated by RFC 2818. I have spent way to much time on this already, that said, you can add a SAN (Subject Alternate Name) by using your own CA! I will cover that later but this does work and I can add sites etc to Netbox.
